File: go/soluc.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	const N = 1000

	A := make([][]float64, N)
	B := make([][]float64, N)
	result := make([][]float64, N)

	for i := 0; i < N; i++ {
		A[i] = make([]float64, N)
		B[i] = make([]float64, N)
		result[i] = make([]float64, N)
	}

	rand.Seed(time.Now().UnixNano())

	for i := 0; i < N; i++ {
		for j := 0; j < N; j++ {
			A[i][j] = rand.Float64()
			B[i][j] = rand.Float64()
		}
	}

	start := time.Now()

	for i := 0; i < N; i++ {
		for j := 0; j < N; j++ {
			for k := 0; k < N; k++ {
				result[i][j] += A[i][k] * B[k][j]
			}
		}
	}

	elapsed := time.Since(start)
	fmt.Printf("Tiempo de ejecución: %s", elapsed)
}

File: rust/soluc.rs
use std::time::Instant;

// Generador aleatorio simple basado en el tiempo
fn simple_random(seed: &mut u64) -> f64 {
    *seed = seed.wrapping_mul(6364136223846793005).wrapping_add(1);
    (*seed % 1000) as f64
}

fn main() {
    const N: usize = 1000;
    let mut seed = 123456789;

    let mut a = vec![vec![0.0; N]; N];
    let mut b = vec![vec![0.0; N]; N];
    let mut result = vec![vec![0.0; N]; N];

    for i in 0..N {
        for j in 0..N {
            a[i][j] = simple_random(&mut seed);
            b[i][j] = simple_random(&mut seed);
        }
    }

    let start = Instant::now();

    for i in 0..N {
        for j in 0..N {
            for k in 0..N {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }

    let duration = start.elapsed();
    println!("Tiempo de ejecución: {:?}", duration);
}
